src/windows/audio.c

```diff
@@ -331,16 +334,20 @@ void MTY_AudioDestroy(MTY_Audio **audio)
 	*audio = NULL;
 }
 
-static uint32_t audio_get_queued_frames(MTY_Audio *ctx)
+static HRESULT audio_get_queued_frames(MTY_Audio *ctx, uint32_t *out)
 {
+	*out = ctx->buffer_size; // indicates that the buffer is full, which also serves
+	                         // as a useful return value in the case of an error.
+
 	if (!ctx->client)
-		return ctx->buffer_size;
+		return AUDCLNT_E_NOT_INITIALIZED;
 
 	UINT32 padding = 0;
-	if (IAudioClient_GetCurrentPadding(ctx->client, &padding) == S_OK)
-		return padding;
+	HRESULT e = IAudioClient_GetCurrentPadding(ctx->client, &padding);
+	if (e == S_OK)
+		*out = padding;
 
-	return ctx->buffer_size;
+	return e;
```

Comment:
Note that theoretically, I could do this:

```c
return IAudioClient_GetCurrentPadding(ctx->client, out);
```

But I think it is better practice to use a local variable when calling a library API like this and only on success, updating the out-param value. This prevents weird cases where a library bug causes the out-param to be filled with something incorrect and causing the app to behave unexpectedly.